vunit/ui.py

```diff
@@ -393,6 +394,41 @@ def add_external_library(self, library_name, path, vhdl_standard=None):
         self._project.add_library(library_name, abspath(path), vhdl_standard, is_external=True)
         return self.library(library_name)
 
+    def add_source_files_from_csv(self, project_csv_path, vhdl_standard=None):
+        """
+        Add a project configuration, mapping all the libraries and files
+        :param project_csv_path: path to csv project specification, each line contains the name 
+                                 of the library and the path to one file 'lib_name,filename'
+        :param vhdl_standard: The VHDL standard used to compile file into this library,
+                              if None, the VUNIT_VHDL_STANDARD environment variable is used
+        :returns: A list of files (:class `.SourceFileList`) that were added
+        """
+        if vhdl_standard is None:
+            vhdl_standard = self._vhdl_standard
+
+        libs_and_files = dict()
+
+        with open(project_csv_path) as f:
+            content = csv.reader(f)
+            for row in content:
+                if len(row) == 2:
+                    lib_name =row[0].strip()
+                    no_normalized_file = row[1].strip()
+                    file_name = normpath(join(dirname(project_csv_path), no_normalized_file))
+                    if lib_name in libs_and_files:
+                        libs_and_files[lib_name] += [file_name]
+                    else:
+                        libs_and_files[lib_name] = [file_name]
+                elif len(row) > 2:
+                    LOGGER.error("More than one library and one file in csv description")
+    
+            list_of_source_files = list()
```

Comment:
= SourceFileList()

Otherwise the return value will be a list() and not a SourceFileList like the documentation claims.